src/mesh/mesh_refinement.cpp

```diff
@@ -95,75 +67,75 @@ void MeshRefinement::RestrictCellCenteredValues(const ParArrayND<Real> &fine,
-        for (int cj = csj; cj <= cej; cj++) {
-          int j = (cj - pmb->cjs) * 2 + pmb->js;
-          pco->CellVolume(k, j, si, ei, fvol_[0][0]);
-          pco->CellVolume(k, j + 1, si, ei, fvol_[0][1]);
-          pco->CellVolume(k + 1, j, si, ei, fvol_[1][0]);
-          pco->CellVolume(k + 1, j + 1, si, ei, fvol_[1][1]);
-          for (int ci = csi; ci <= cei; ci++) {
-            int i = (ci - pmb->cis) * 2 + pmb->is;
-            // KGF: add the off-centered quantities first to preserve FP symmetry
-            Real tvol = ((fvol_[0][0](i) + fvol_[0][1](i)) +
-                         (fvol_[0][0](i + 1) + fvol_[0][1](i + 1))) +
-                        ((fvol_[1][0](i) + fvol_[1][1](i)) +
-                         (fvol_[1][0](i + 1) + fvol_[1][1](i + 1)));
-            // KGF: add the off-centered quantities first to preserve FP symmetry
-            coarse(n, ck, cj, ci) =
-                (((fine(n, k, j, i) * fvol_[0][0](i) +
-                   fine(n, k, j + 1, i) * fvol_[0][1](i)) +
-                  (fine(n, k, j, i + 1) * fvol_[0][0](i + 1) +
-                   fine(n, k, j + 1, i + 1) * fvol_[0][1](i + 1))) +
-                 ((fine(n, k + 1, j, i) * fvol_[1][0](i) +
-                   fine(n, k + 1, j + 1, i) * fvol_[1][1](i)) +
-                  (fine(n, k + 1, j, i + 1) * fvol_[1][0](i + 1) +
-                   fine(n, k + 1, j + 1, i + 1) * fvol_[1][1](i + 1)))) /
-                tvol;
-          }
-        }
-      }
-    }
+    pmb->par_for(
+        "RestrictCellCenteredValues3d", sn, en, csk, cek, csj, cej, csi, cei,
+        KOKKOS_LAMBDA(const int n, const int ck, const int cj, const int ci) {
+          int k = (ck - cks) * 2 + ks;
+          int j = (cj - cjs) * 2 + js;
+          int i = (ci - cis) * 2 + is;
+          // KGF: add the off-centered quantities first to preserve FP symmetry
+          const Real vol000 = coords.Volume(k, j, i);
+          const Real vol001 = coords.Volume(k, j, i + 1);
+          const Real vol010 = coords.Volume(k, j + 1, i);
+          const Real vol011 = coords.Volume(k, j + 1, i + 1);
+          const Real vol100 = coords.Volume(k + 1, j, i);
+          const Real vol101 = coords.Volume(k + 1, j, i + 1);
+          const Real vol110 = coords.Volume(k + 1, j + 1, i);
+          const Real vol111 = coords.Volume(k + 1, j + 1, i + 1);
+          Real tvol = ((vol000 + vol010) + (vol001 + vol011)) +
+                      ((vol100 + vol110) + (vol101 + vol111));
+          // KGF: add the off-centered quantities first to preserve FP symmetry
+          coarse(n, ck, cj, ci) =
+              (((fine(n, k, j, i) * vol000 + fine(n, k, j + 1, i) * vol010) +
+                (fine(n, k, j, i + 1) * vol001 + fine(n, k, j + 1, i + 1) * vol011)) +
+               ((fine(n, k + 1, j, i) * vol100 + fine(n, k + 1, j + 1, i) * vol110) +
+                (fine(n, k + 1, j, i + 1) * vol101 +
+                 fine(n, k + 1, j + 1, i + 1) * vol111))) /
+              tvol;
```

Comment:
a little late now, but I was going to suggest aligning these and wrapping them in `// clang-format off`, `// clang-format on` blocks.